devicetypes/erocm123/inovelli-2-channel-smart-plug-mcd.src/inovelli-2-channel-smart-plug-mcd.groovy

```diff
@@ -0,0 +1,305 @@
+		fingerprint manufacturer: "0312", prod: "0221", model: "611C", deviceJoinName: "Inovelli Outlet" // Inovelli 2-Channel Outdoor Smart Plug
+		fingerprint manufacturer: "015D", prod: "0221", model: "611C", deviceJoinName: "Inovelli Outlet" // Inovelli 2-Channel Outdoor Smart Plug
+		fingerprint manufacturer: "015D", prod: "6100", model: "6100", deviceJoinName: "Inovelli Outlet" // Inovelli 2-Channel Outdoor Smart Plug
+		fingerprint manufacturer: "0312", prod: "6100", model: "6100", deviceJoinName: "Inovelli Outlet" // Inovelli 2-Channel Outdoor Smart Plug
+		fingerprint manufacturer: "015D", prod: "2500", model: "2500", deviceJoinName: "Inovelli Outlet" // Inovelli 2-Channel Smart Plug w/Scene
+	}
+	simulator {}
+	preferences {}
+	tiles {
+		multiAttributeTile(name: "switch", type: "lighting", width: 6, height: 4, canChangeIcon: true) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "off", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "on", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc", nextState: "turningOff"
+				attributeState "turningOff", label: '${name}', action: "switch.on", icon: "st.switches.switch.off", backgroundColor: "#ffffff", nextState: "turningOn"
+				attributeState "turningOn", label: '${name}', action: "switch.off", icon: "st.switches.switch.on", backgroundColor: "#00a0dc", nextState: "turningOff"
+			}
+		}
+		standardTile("refresh", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+		main(["switch"])
+		details(["switch",
+				 childDeviceTiles("all"), "refresh"
+		])
+	}
+}
+
+def parse(String description) {
+	def result = []
+	def cmd = zwave.parse(description)
+
+	if (cmd) {
+		result += zwaveEvent(cmd)
+		log.debug "Parsed ${cmd} to ${result.inspect()}"
+	} else {
+		log.debug "Non-parsed event: ${description}"
+	}
+
+	return result
+}
+
+def handleSwitchEndpointEvent(cmd, ep) {
+	def event
+
+	def childDevice = childDevices.find {
+			it.deviceNetworkId == "$device.deviceNetworkId:$ep"
+		}
+	childDevice?.sendEvent(name: "switch", value: cmd.value ? "on" : "off")
+
+	if (cmd.value) {
+		event = [createEvent([name: "switch", value: "on"])]
+	} else {
+		def allOff = true
+
+		childDevices.each { n ->
+				if (n.currentState("switch")?.value != "off")
+					allOff = false
+			}
+
+		if (allOff) {
+			event = [createEvent([name: "switch", value: "off"])]
+		} else {
+			event = [createEvent([name: "switch", value: "on"])]
+		}
+	}
+
+	return event
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "BasicReport ${cmd} - outlet ${ep}"
+
+	if (ep) {
+		return handleSwitchEndpointEvent(cmd, ep)
+	}
```

Comment:
Assuming this device never sends Basic Report from the root device?

Reply:
I ask because I generally would recommend to have the same handler for incoming Basic Set and Report

Reply:
@kianooshST That's my assumption. Since I don't have a device on hand and this is the logic from the original code I left it as part of my consolidation. We can always update it later.